From 08d47ad147b51f18418f180845247a380f076347 Mon Sep 17 00:00:00 2001
From: Anson Huang <b20788@freescale.com>
Date: Wed, 6 Jun 2012 09:36:09 +0800
Subject: [PATCH 1573/2463] ENGR00212490 [SMP]Fix build error of calibrate
 when no define SMP

Fix build error, only under SMP enviroment, we use per_cpu cpu_data
to decide to redo calibrate delay or skip.

Signed-off-by: Anson Huang <b20788@freescale.com>
---
 init/calibrate.c |    7 ++++++-
 1 files changed, 6 insertions(+), 1 deletions(-)

diff --git a/init/calibrate.c b/init/calibrate.c
index ebab7b3..a4f57ff 100644
--- a/init/calibrate.c
+++ b/init/calibrate.c
@@ -249,13 +249,16 @@ void __cpuinit calibrate_delay(void)
 {
 	unsigned long lpj;
 	static bool printed;
+#ifdef CONFIG_SMP
 	int this_cpu = smp_processor_id();
-
 	if (per_cpu(cpu_data, this_cpu).loops_per_jiffy) {
 		lpj = per_cpu(cpu_data, this_cpu).loops_per_jiffy;
 		pr_info("Calibrating delay loop (skipped) "
 				"already calibrated this CPU");
 	} else if (preset_lpj) {
+#else
+	if (preset_lpj) {
+#endif
 		lpj = preset_lpj;
 		if (!printed)
 			pr_info("Calibrating delay loop (skipped) "
@@ -273,7 +276,9 @@ void __cpuinit calibrate_delay(void)
 			pr_info("Calibrating delay loop... ");
 		lpj = calibrate_delay_converge();
 	}
+#ifdef CONFIG_SMP
 	per_cpu(cpu_data, this_cpu).loops_per_jiffy = lpj;
+#endif
 	if (!printed)
 		pr_cont("%lu.%02lu BogoMIPS (lpj=%lu)\n",
 			lpj/(500000/HZ),
-- 
1.7.7.4

